File: 정수빈/Programmers/Level2/이진 변환 반복하기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/70129
# 설명
# 1. 0을 제거한다.
# 2. 0을 제거한 문자열의 길이를 2진법으로 변환한다.
# 3. 이진변환 한 횟수와 지워진 0의 개수를 센다.

def solution(s):
    answer = []
    count = 0
    zero_count = 0
    while s != "1":
        reduce_s = s.replace("0", "")
        y_s = format(len(reduce_s), 'b')
        zero_count += len(s) - len(reduce_s)
        count += 1
        s = y_s
    return [count, zero_count]

File: 곽경진/Programmers/Level2/이진_변환_반복하기.java
/*
이진 변환 반복하기 
https://school.programmers.co.kr/learn/courses/30/lessons/70129
    1. 0 과 1로 이루어진 문자열의 0을 모두제거 
    2. 그문자열의 0을 제거 
    3. 길이를 2진수로 변환
*/

class Solution {
    public int[] solution(String s) {
        int count = 0;
        int length = 0;
        while(true){
            int tempLength = s.length();
            String tempString = s.replace("0","");
            length += tempLength - tempString.length();
            count++;
            if(tempString.length() == 1){
                break;
            }
            s = Integer.toBinaryString(tempString.length());
        }
        int[] answer = {count,length};
        return answer;
    }
}